pkg/nas/nodeserver_test.go

```diff
@@ -0,0 +1,202 @@
+package nas
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"testing"
+
+	"github.com/container-storage-interface/spec/lib/go/csi"
+	"github.com/kubernetes-sigs/alibaba-cloud-csi-driver/pkg/cnfs/v1beta1"
+	"github.com/kubernetes-sigs/alibaba-cloud-csi-driver/pkg/nas/internal"
+	"github.com/kubernetes-sigs/alibaba-cloud-csi-driver/pkg/utils"
+	"github.com/stretchr/testify/assert"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/tools/record"
+)
+
+type fakeCNFSGetter struct {
+	cnfsMap map[string]*v1beta1.ContainerNetworkFileSystem
+}
+
+func newFakeCNFSGetter(cnfsList ...*v1beta1.ContainerNetworkFileSystem) *fakeCNFSGetter {
+	cnfsMap := make(map[string]*v1beta1.ContainerNetworkFileSystem)
+	for _, cnfs := range cnfsList {
+		if cnfs != nil {
+			cnfsMap[cnfs.Name] = cnfs
+		}
+	}
+	return &fakeCNFSGetter{cnfsMap}
+}
+
+func (f *fakeCNFSGetter) GetCNFS(_ context.Context, name string) (*v1beta1.ContainerNetworkFileSystem, error) {
+	if cnfs, ok := f.cnfsMap[name]; ok {
+		return cnfs, nil
+	}
+	return nil, fmt.Errorf("CNFS %s not found", name)
+}
+
+func startListeningFor(addr string) {
+	listener, err := net.Listen("tcp", addr)
+	if err != nil {
+		panic(err)
+	}
+	defer listener.Close()
+
+	for {
+		conn, _ := listener.Accept()
+		if conn != nil {
+			conn.Close()
+		}
+	}
```

Comment:
This will not terminate after the test case finishes. How about making use of `t.Cleanup()`?